### ISSUE TITLE
Reset Password Confirmation Link Email Not Store Scoped For Global Customers

### ISSUE THREAD

**Issue by floorz** [magento/magento2#23295]
### Preconditions (*)

1. 2.2.4
2. 2.2.5
3. 2.2.6
4. 2.2.7
5. 2.2.8
6. 2.3.0
7. 2.3.1

### Steps to reproduce (*)

1. Setup multi-store site with at least 2 websites
2. Configure customer accounts to be shared globally from admin
3. Create a customer account associated to the second website
4. Initiate reset password email via rest api at *PUT /V1/customers/password*
Payload:
```
{
	"email": "user@example.com",
	"template": "email_reset"
}
```
5. Receive reset password email from the default store, not the second store

### Expected result (*)

1. Receive email from the store the customer is assigned to

### Actual result (*)

1. Received email from the initial default store and not the store the customer is assigned to

### Use Case Information
We have a client on Magento Cloud with separate websites setup for each region/language (US, Canada, United Kingdom, Spain, China, Japan, etc.). They have mobile apps that will allow customers to register, sign in, register products (custom), and initiate reset password. We needed separate websites to support different base currencies per website.

The mobile apps don't know which website url to use until a customer authenticates which caused a chicken/egg situation with authentication/account creation. We made customer accounts global so the mobile apps can hit the default website url for authentication and be scoped based on the customer's token/assigned store.

The reset password endpoint appears to ignore the customer's store and use the store derived from the website url scoping.

### Investigation
In a couple of other methods in the same `\Magento\Customer\Model\EmailNotification` class, the customer's assigned store id is used instead of the store derived from website domain/url scoping:

https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Customer/Model/EmailNotification.php#L169-L172
https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Customer/Model/EmailNotification.php#L195-L198
https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Customer/Model/EmailNotification.php#L220-L223
https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Customer/Model/EmailNotification.php#L319-L322

This seems to be the only method not using customer's store id, and the one causing this issue:
https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Customer/Model/EmailNotification.php#L343-L346

I'm sort of confused about which approach is the correct approach on a broader level, relying on the website domain/url for determining website scope, or the data passed in being website aware like in the case of the customer.

**Comment by floorz** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @floorz. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @floorz, here is your [Magento instance](https://i-23295-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23295-2-3-develop.instances.magento-community.engineering/admin](https://i-23295-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by floorz** [comment]
Could not reproduce on the vanilla instance provided as emails don't seem to send from there. Did reproduce locally.

**Comment by syed92hasan** [comment]
I am working on this at #MM19UK

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-17601`, `MC-17602` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by rdudhara** [comment]
I am working on it.

**Comment by LucieAAY** [comment]
Hello,

I'm currently experiencing the same problem on version 2.2.8, do you have any news about its resolution?

**Comment by magento-engcom-team** [comment]
Hi @floorz. Thank you for your report. 
The issue has been fixed in magento/magento2#24783 by @kisroman in 2.3-develop branch
Related commit(s):
 - [35ad339af8373cb2280a473f070c57a98da3faee](https://github.com/magento/magento2/commit/35ad339af8373cb2280a473f070c57a98da3faee)

The fix will be available with the upcoming 2.3.4 release.
